### ISSUE TITLE
Using Non-English language for FastText  text classification 

### ISSUE THREAD

**Issue by iamyihwa** [facebookresearch/fastText#683]
Hello, 
I am new to fast text. 
For training Fasttext for text classification (https://fasttext.cc/docs/en/supervised-tutorial.html) can I use wordvectors that are not english as defined in https://fasttext.cc/docs/en/crawl-vectors.html ? 

I might have missed something but in the command for training the text classifier I didn't see a way to change the word vector.. 

Thanks in advance!

**Comment by Celebio** [comment]
Hi @iamyihwa ,
You can use the classification in any language, you are not required to specify word vectors.

Regards,
Onur

**Comment by Celebio** [comment]
Hi @iamyihwa,
Actually there is not a language detection per se. The classification task builds a dictionary based on the ngrams (subwords) of your dataset, so your training set could be actually in any language (even an invented one), or any other discrete representation.

You can have a look at the [`supervised`](https://github.com/facebookresearch/fastText/blob/master/src/fasttext.cc#L337) function and follow the execution to see what happens from there.

Regards,
Onur

**Comment by Celebio** [comment]
Hi @iamyihwa ,
How many samples and how many labels do you have? What are the values of your hyperparameters?

Onur

**Comment by iamyihwa** [comment]
Hi @celebio, 
The parameters I used were :  
model = train_supervised(
        input=train_data, epoch=25, lr=1.0, wordNgrams=2, verbose=2, minCount=1
    )

I have 10026 sentences that i used for training.

**Comment by EdouardGrave** [comment]
Hi @iamyihwa,

Since you have a small training set, using pre-trained word vectors (instead of learning these from scratch on your training set), might give better results. This can be done with fastText using the `pretrainedVectors` option. You can use the Spanish word vectors which were trained on the common crawl data, available here: https://fasttext.cc/docs/en/crawl-vectors.html.

Best,
Edouard.

**Comment by iamyihwa** [comment]
Hi @EdouardGrave 
Thanks for getting back to this issue. 
Yes I know of the fasttext word embeddings, and I have used in another project. 
However [in this tutorial](https://fasttext.cc/docs/en/supervised-tutorial.html), I don't see a way to add pretrainedVectors. 
Are you referring to any other method of adding them? Could you kindly direct me to that page? 
Thanks in advance!

**Comment by EdouardGrave** [comment]
Hi @iamyihwa,

We have not updated the documentation to reflect the usage of pre-trained word vectors yet. The way to use it is to give the path to the file containing the word vectors, e.g.
```
$ ./fasttext supervised -input train.txt -output model -dim 300 -pretrainedVectors cc.es.300.vec
```
Please note that for now, the pre-trained vectors have to be stored in text format. Hope this answer your questions!

Best,
Edouard

**Comment by masonreznov** [comment]
Can similar language pretained wieghts be used if the pretrained model for that language is not available?

**Comment by azamatolegen** [comment]
> Hi @iamyihwa,
> 
> We have not updated the documentation to reflect the usage of pre-trained word vectors yet. The way to use it is to give the path to the file containing the word vectors, e.g.
> 
> ```
> $ ./fasttext supervised -input train.txt -output model -dim 300 -pretrainedVectors cc.es.300.vec
> ```
> 
> Please note that for now, the pre-trained vectors have to be stored in text format. Hope this answers your questions!
> 
> Best,
> Edouard

thanks, I looked for it whole day, it should be in documentation.
